rdrf/rdrf/patient_view.py

```diff
@@ -712,14 +717,20 @@ def post(self, request, registry_code, patient_id):
         context["archive_patient_url"] =  patient.get_archive_url(registry_model) if request.user.can_archive else ""
         context["consent"] = consent_status_for_patient(registry_code, patient)
 
-        if request.user.is_parent:
-            context['parent'] = ParentGuardian.objects.get(user=request.user)
+
+        if "section_blacklist" in registry_model.metadata:
+            section_blacklist = registry_model.metadata["section_blacklist"]
+            section_blacklist = [_("{}".format(x)) for x in section_blacklist]
+        else:
+            section_blacklist = []
+
+        context["section_blacklist"] = section_blacklist
```

Comment:
avoid duplicating code - 
extract a function of the registry model

Reply:
Fixed.